### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Mahnoor-Faisal0303**

### CODE REVIEW

File: src/Component/AddButtonTodo.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+interface AddButtonTodoProps {
+  todoText: string;
+  setShowAlert: React.Dispatch<React.SetStateAction<boolean>>;
+  setTodoText: React.Dispatch<React.SetStateAction<string>>;
+}
+
+const AddButtonTodo: React.FC<AddButtonTodoProps> = ({ todoText, setShowAlert, setTodoText }) => {
+  const dispatch = useDispatch();
+
+  const handleAddButtonClick = () => {
```
Comment (saad277) [inline src/Component/AddButtonTodo.tsx]: move all this logic in a custom hook

File: src/Component/AddButtonTodo.tsx
Comment (saad277) [inline src/Component/AddButtonTodo.tsx:1]: in components there should be Button only

File: src/Component/InputTextTodo.tsx
Comment (saad277) [inline src/Component/InputTextTodo.tsx:1]: component should be Input only

File: src/Component/LogoutTodoButton.tsx
Comment (saad277) [inline src/Component/LogoutTodoButton.tsx:1]: component should be Button only in this folder

File: src/Component/UpdateButtonTodo.tsx
Comment (saad277) [inline src/Component/UpdateButtonTodo.tsx:1]: component should be button only with logic coming from the parent

File: src/Pages/Home.tsx [hunk 2]
```diff
@@ -1,16 +1,86 @@
...
+  const [showAlert, setShowAlert] = useState<boolean>(false);
+  const [showUpdateButton, setShowUpdateButton] = useState<boolean>(false);
+  const [showAddButton, setShowAddButton] = useState<boolean>(true);
+  const dispatch = useDispatch();
+  const todoList = useSelector((state: RootState) => state.todos.todoList);
+
+  const handleDeleteButtonClick = (id: string) => {
+    dispatch(deleteTodo(id));
+  };
+  const [newId, setNewId] = useState<string>("");
```
Comment (saad277) [inline src/Pages/Home.tsx]: i dont think we need this , we can just use uuid

File: src/Pages/Home.tsx [hunk 3]
```diff
@@ -1,16 +1,86 @@
...
+              </ListItemSecondaryAction>
+            </ListedItem>
+          ))}
+
+        </TodoList>
+
+        {
+          showAlert && (
+            <Alert variant="filled" severity="info">
+              Please write something!
```
Comment (saad277) [inline src/Pages/Home.tsx]: handle this with hook form

File: src/Pages/LoginScreen.tsx [hunk 4]
```diff
@@ -1,111 +1,134 @@
...
         event.preventDefault();
     };
 
-    const handleKeyPress = (event: React.KeyboardEvent<unknown>, nextInputRef: React.RefObject<HTMLInputElement>) => {
-        if (event.key === 'Enter') {
-            if (nextInputRef && nextInputRef.current) {
-                nextInputRef.current.focus();
-            }
+    useEffect(() => {
+        const loginData = localStorage.getItem("loggedInData");
```
Comment (saad277) [inline src/Pages/LoginScreen.tsx]: why do we need this when we have redux persist

File: src/Pages/LoginScreen.tsx [hunk 5]
```diff
@@ -1,111 +1,134 @@
...
+    password: string
+}
 
 const LoginScreen: React.FC = () => {
+    const {
+        register,
+        formState: { errors },
+        handleSubmit,
+        setError,
+    } = useForm<IFormInput>()
```
Comment (saad277) [inline src/Pages/LoginScreen.tsx:25]: use zod validation with hook form

File: src/Pages/LoginScreen.tsx [hunk 6]
```diff
@@ -1,111 +1,134 @@
...
+            const isUserValid = newData.find((userData: { password: string; email: string; }
+            ) => userData.email === userEnteredEmail && userData.password === userEnteredPassword);
 
-        } else {
-            setShowAlert(true);
-            setTimeout(function () {
-                setShowAlert(false);
-            }, 2000);
+            if (isUserValid) {
+                localStorage.setItem('loggedInData', JSON.stringify(userEnteredEmail));
```
Comment (saad277) [inline src/Pages/LoginScreen.tsx]: no need for this when we have redux persist

File: src/Pages/SignupScreen.tsx [hunk 7]
```diff
@@ -1,55 +1,38 @@
...
+    confirmPassword: string
+}
+const SignupScreen: React.FC = () => {
 
+    const {
+        register,
+        formState: { errors },
+        handleSubmit,
+        setError,
+    } = useForm<IFormInput>()
```
Comment (saad277) [inline src/Pages/SignupScreen.tsx]: use zod validation

File: src/Pages/Home.tsx (hunk 2 above)
Comment (Mahnoor-Faisal0303) [reply src/Pages/Home.tsx]: it is just used for storing uuid of list, for update and delete use this stored uuid,

File: src/Component/AddButtonTodo.tsx
Comment (saad277) [inline src/Component/AddButtonTodo.tsx:1]: rename to Button

File: src/Component/AddButtonTodo.tsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+import { _Button } from '../Style/HomeStyle';
+
+interface AddButtonProps {
+ // onClick: () => void;
+  type: "button" | "submit" | "reset" | undefined;
+}
+
+const AddButton: React.FC<AddButtonProps> = ({ type }) => {
+  return (
+    <_Button type={type} >ADD</_Button>
```
Comment (saad277) [inline src/Component/AddButtonTodo.tsx]: should accept title as prop

File: src/Component/DeleteButton.tsx
Comment (saad277) [inline src/Component/DeleteButton.tsx:1]: there should be only 1 button component

File: src/Component/EditButton.tsx
Comment (saad277) [inline src/Component/EditButton.tsx:1]: there should be only 1 button component

File: src/Component/LogoutTodoButton.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
+// import { generatePath, useNavigate } from "react-router-dom";
```
Comment (saad277) [inline src/Component/LogoutTodoButton.tsx]: remove commented code

File: src/Component/UpdateButtonTodo.tsx
Comment (saad277) [inline src/Component/UpdateButtonTodo.tsx:1]: there should be only 1 button component

File: src/Component/LogoutTodoButton.tsx
Comment (saad277) [inline src/Component/LogoutTodoButton.tsx:1]: there should be only 1 button component

File: src/Pages/Home.tsx [hunk 10]
```diff
@@ -1,16 +1,125 @@
...
+  const todoList = useSelector((state: RootState) => state.todos.todoList);
+  const navigate = useNavigate();
+
+  const handleDeleteButtonClick = (id: string) => {
+    dispatch(deleteTodo(id));
+  };
+
+  const [newId, setNewId] = useState<string>("");
+
+  const handleEditButtonClick = (id: string) => {
```
Comment (saad277) [inline src/Pages/Home.tsx]: move this to a custom hook as file is getting long

File: src/Store/Slices/todoSlice.tsx [hunk 11]
```diff
@@ -7,13 +7,14 @@ interface TodoState {
     editText: { id: string; name: string } | null;
 
 }
+const initialState: TodoState = {
+    todoList: [],
+    editText: null,
```
Comment (saad277) [inline src/Store/Slices/todoSlice.tsx]: why is edit text here ?

**Pull request merged**